tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,140 @@
+    //3. In the list of pets, locate the pet with a name "Samantha" and click "Add Visit" button
+    const addVisitButtonOwnerPage = page.locator('app-pet-list', { hasText: 'Samantha' })
+    await addVisitButtonOwnerPage.getByText('Add Visit').click()
+    //4. Add the assertion that "New Visit" is displayed as header of the page
+    await expect(page.getByRole("heading")).toHaveText("New Visit")
+    //5. Add the assertion that pet name is "Samantha" and owner name is "Jean Coleman"
+    await expect(page.locator('tr', { hasText: 'Samantha' })).toContainText('Jean Coleman')
+    //6. Click on the calendar icon and select the current date in date picker
+    const calendarIcon = page.getByLabel('Open calendar')
+    await calendarIcon.click()
+    await page.locator('span.mat-calendar-body-today').click()
+    //7. Add assertion that date is displayed in the format "YYYY/MM/DD"
+    const date = new Date()
+    const year = date.getFullYear();
+    const month = date.toLocaleString('en-US', { month: '2-digit' })
+    const day = date.toLocaleString('en-US', { day: '2-digit' })
+    const expectedDermatologyAppointmentDate = `${year}-${month}-${day}`
+    function formatDate(date: Date, expectedFormat: string): string {
+        if (!expectedFormat) {
+            throw new Error('The "expectedFormat" parameter is required. Please specify a valid date format.')
+        }
+        switch (expectedFormat) {
+            case 'YYYY/MM/DD':
+                return `${year}/${month}/${day}`
+            case 'YYYY-MM-DD':
+                return `${year}-${month}-${day}`
+            case 'MM YYYY':
+                return `${month} ${year}`;
+            default:
+                throw new Error('The "expectedFormat" parameter is required. Please specify a valid date format.')
+        }
+    }
+
+    const newVisitPageDateFormat = formatDate(date, 'YYYY/MM/DD')
+    const ownerPageDateFormat = formatDate(date, 'YYYY-MM-DD')
+
+    console.log('Formatted Date on New Visit Page:', newVisitPageDateFormat);
+    console.log('Formatted Date on Owner Page:', ownerPageDateFormat);
+
+    //8. Type the description in the field, for example, "dermatologists visit" and click "Add Visit" button
+    await page.locator('#description').fill('veterinary internist visit')
+    await page.getByRole('button', { name: 'Add Visit' }).click()
+
+    //9. Add assertion that selected date of visit is displayed at the top of the list of visits 
+    //for "Samantha" pet on the "Owner Information" page and is in the format "YYYY-MM-DD"
+    await expect(addVisitButtonOwnerPage.locator('app-visit-list tr td').first()).toHaveText(expectedDermatologyAppointmentDate)
+
+    //10. Add one more visit for "Samantha" pet by clicking "Add Visit" button
+    await addVisitButtonOwnerPage.getByText('Add Visit').click()
+    //11. Click on the calendar icon and select the date which is 45 days back from the current date
+    await page.locator('.mdc-icon-button').click()
```

Comment:
Earlier in your script, you used a more "readable" locator for this web element. Keep consistency in the scripting.

Reply:
When I'm deeply focused on my code, I start strong, but then something happens, and I lose my concentration and stumble. 😄

Reply:
I did ))))) I was commenting and I forgot.